### PR TITLE
Issue/16150 blogging prompt list of respondents

### LINKED ISSUE #16150: [Blogging prompts v1] Implement Prompt Card scaffolding in "My Site"

### PR DESCRIPTION
**Pull request by khaykov**
Fixes #16150

This PR replaces the mock list of prompt respondents with an existing piece of code used to display users who like a blog post.

List of additional:
- I extracted "train of face" code into a separate package (`org.wordpress.android.ui.avatars`) and renamed it to something a bit more generic.
- Added support for separate label color in trailing text label.
- Plurals support.
- Updated the code that determines the width of the text label to work with any container. We need to manually set the width of that label, since it is located in RecyclerView and given the chance it will extend horizontally, instead of wrapping.
- List of `BloggingPromptRespondent` is now included in blogging prompt model.
- `BloggingPromptsOnboardingDialogFragment` is using the new adapter with some dummy data. Not sure at the moment what we will actually show there.

PR looks a bit big, but most of the content in `org.wordpress.android.ui.avatars` package is pre-existing. For some reason some of those files are shown as new.

1 answer:
[Image from Gyazo](https://gyazo.com/0368fb2ab1d8f51c260e5faebe323dde)

5 answers:
[Image from Gyazo](https://gyazo.com/a4d54e4bf253837cf4a3b30865ca7872)

No Answers:
[Image from Gyazo](https://gyazo.com/bacb4858333c0fe70add2a41591ab8fd)

To test:
- Confirm that the blogging prompt respondents list looks as expected in both My Site card and Feature Dialog.
- Confirm that the list of likers looks the same in Post Details.

## Regression Notes
1. Potential unintended areas of impact
- "Train of Likers" in post detail might looks off.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Visual confirmation and existing tests.

3. What automated tests I added (or what prevented me from doing so)
BP card builder test.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/onboarding/BloggingPromptsOnboardingDialogFragment.kt [hunk 1]
```diff
@@ -71,7 +83,23 @@ class BloggingPromptsOnboardingDialogFragment : FeatureIntroductionDialogFragmen
             contentTop.text = getString(readyState.contentTopRes)
             cardCoverView.setOnClickListener { /*do nothing*/ }
             promptCard.promptContent.text = getString(readyState.promptRes)
-            promptCard.numberOfAnswers.text = getString(readyState.answersRes, readyState.answersCount)
+
+            val layoutManager = FlexboxLayoutManager(context)
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/onboarding/BloggingPromptsOnboardingDialogFragment.kt]: NP: would it make sense to use `apply` here?

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloggingprompts/BloggingPromptCardViewHolder.kt [hunk 2]
```diff
@@ -47,15 +52,36 @@ class BloggingPromptCardViewHolder(
             )
         }
         uiHelpers.updateVisibility(answeredPromptControls, card.isAnswered)
+
+        val layoutManager = FlexboxLayoutManager(answeredUsersRecycler.context)
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloggingprompts/BloggingPromptCardViewHolder.kt]: NP: would it make sense to use `apply` here?

**Review by RenanLukas — APPROVED** [review]
Thanks for the PR @khaykov . It works as expected :shipit: 
I've left a couple of completely optional nitpicks

**Comment by khaykov** [comment]
Thanks for taking a look, @RenanLukas ! Implemented your suggestions 👍

**Pull request merged**